src/model_signing/_cli.py

```diff
@@ -0,0 +1,451 @@
+        removing anything.
+        """
+        if args[0] in self._supported_modes:
+            return super().resolve_command(ctx, args)
+        _, cmd, _ = super().resolve_command(ctx, args)
+        return cmd.name, cmd, args
+
+
+@click.group(
+    context_settings=dict(help_option_names=["-h", "--help"]),
+    epilog=(
+        "Check https://sigstore.github.io/model-transparency for "
+        "documentation and more details."
+    ),
+)
+@click.version_option(model_signing.__version__, "--version")
+def main() -> None:
+    """ML model signing and verification.
+
+    Use each subcommand's `--help` option for details on each mode.
+    """
+
+
+@main.group(name="sign", subcommand_metavar="PKI_METHOD", cls=PKICmdGroup)
+def _sign() -> None:
+    """Sign models.
+
+    Produces a cryptographic signature (in the form of a Sigstore bundle) for a
+    model. We support any model format, either as a single file or as a
+    directory.
+
+    We support multiple PKI methods, specified as subcommands. By default, the
+    signature is generated via Sigstore (as if invoking `sigstore` subcommand).
+
+    Use each subcommand's `--help` option for details on each mode.
+    """
+
+
+@_sign.command(name="sigstore")
+@_model_path_argument
+@_write_signature_option
+@click.option(
+    "--use_ambient_credentials",
+    type=bool,
+    is_flag=True,
+    help="Use credentials from ambient environment.",
+)
+@click.option(
+    "--identity_token",
+    type=str,
+    metavar="TOKEN",
+    help=(
+        "Fixed OIDC identity token to use instead of obtaining credentials "
+        "from OIDC flow or from the environment."
+    ),
+)
+@click.option(
+    "--use_staging",
+    type=bool,
+    is_flag=True,
+    help="Use Sigstore's staging instance.",
+)
+def _sign_sigstore(
+    model_path: pathlib.Path,
+    signature: pathlib.Path,
+    use_ambient_credentials: bool,
+    use_staging: bool,
+    identity_token: Optional[str] = None,
+) -> None:
+    """Sign using Sigstore (DEFAULT signing method).
+
+    Signing the model at MODEL_PATH, produces the signature at SIGNATURE_PATH
+    (as per `--signature` option).
+
+    If using Sigstore, we need to provision an OIDC token. In general, this is
+    taken from an interractive OIDC flow, but ambient credentials could be used
+    to use workload identity tokens (e.g., when running in GitHub actions).
+    Alternatively, a constant identity token can be provided via
+    `--identity_token`.
+
+    Sigstore allows users to use a staging instance for test-only signatures.
+    Passing the `--use_staging` flag would use that instance instead of the
+    production one.
+    """
+    signer = sigstore.SigstoreDSSESigner(
+        use_ambient_credentials=use_ambient_credentials,
+        use_staging=use_staging,
+        identity_token=identity_token,
+    )
+    _serialize_and_sign(model_path, signer, signature)
+
+
+@_sign.command(name="key")
```

Comment:
similarly this was `private-key`

Reply:
Yeah, since we want to match with verification which uses public key actually